From b5f423473824205664a85f08face91e4141805fc Mon Sep 17 00:00:00 2001
From: "Sreeves, Francis" <francis.sreeves@digi.com>
Date: Tue, 15 Jun 2021 10:45:43 +0100
Subject: [PATCH 10/10] myproposal: remove weak algorithms from SSH server

As part of FirstNet security requirements, remove the following
algorithms from the SSH server's default proposal:

MACs:

    umac-64-etm@openssh.com
    hmac-sha1-etm@openssh.com
    umac-64@openssh.com
    hmac-sha1

KexAlgorithms:

    diffie-hellman-group16-sha512
    diffie-hellman-group18-sha512
    diffie-hellman-group14-sha256

DAL-4797
---
 myproposal.h | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/myproposal.h b/myproposal.h
index 6834854..2732c47 100644
--- a/myproposal.h
+++ b/myproposal.h
@@ -31,13 +31,14 @@
 	"ecdh-sha2-nistp256," \
 	"ecdh-sha2-nistp384," \
 	"ecdh-sha2-nistp521," \
-	"diffie-hellman-group-exchange-sha256," \
+	"diffie-hellman-group-exchange-sha256"
+
+#define KEX_CLIENT_KEX \
+	KEX_SERVER_KEX "," \
 	"diffie-hellman-group16-sha512," \
 	"diffie-hellman-group18-sha512," \
 	"diffie-hellman-group14-sha256"
 
-#define KEX_CLIENT_KEX KEX_SERVER_KEX
-
 #define	KEX_DEFAULT_PK_ALG	\
 	"ssh-ed25519-cert-v01@openssh.com," \
 	"ecdsa-sha2-nistp256-cert-v01@openssh.com," \
@@ -64,18 +65,19 @@
 #define KEX_CLIENT_ENCRYPT KEX_SERVER_ENCRYPT
 
 #define	KEX_SERVER_MAC \
-	"umac-64-etm@openssh.com," \
 	"umac-128-etm@openssh.com," \
 	"hmac-sha2-256-etm@openssh.com," \
 	"hmac-sha2-512-etm@openssh.com," \
-	"hmac-sha1-etm@openssh.com," \
-	"umac-64@openssh.com," \
 	"umac-128@openssh.com," \
 	"hmac-sha2-256," \
-	"hmac-sha2-512," \
-	"hmac-sha1"
+	"hmac-sha2-512"
 
-#define KEX_CLIENT_MAC KEX_SERVER_MAC
+#define KEX_CLIENT_MAC \
+	KEX_SERVER_MAC "," \
+	"umac-64-etm@openssh.com," \
+	"hmac-sha1-etm@openssh.com," \
+	"umac-64@openssh.com," \
+	"hmac-sha1"
 
 /* Not a KEX value, but here so all the algorithm defaults are together */
 #define	SSH_ALLOWED_CA_SIGALGS	\
-- 
2.30.2

